src/palace.pyx

```diff
@@ -363,17 +364,19 @@ cdef class Context:
         self.device = device
         self.listener = Listener(self)
         self.message_handler = MessageHandler()
+        self.previous = <alure.Context> nullptr
```

Comment:
I'm not sure if this is necessary:  `__exit__` should never be called without `__enter__` previously.  If it is, let the error be reported as implementation detail.

Reply:
Alright then. I just felt that every attributes should be initialized with some value.

Reply:
In most cases, yes, but the context manager is not warrantied to be used.  Normally in Python such temporary attributes are left undocumented, but C/C++/Cython binds early, and I guess this is the clearest we can do.